scripts/verify_security_fixes.py

```
@@ -116,12 +116,49 @@ def oversized_generator():
+    resolved = generate_embeddings.resolve_fetch_url(valid_url, replacement_base, target_base)
+    if resolved == "http://localhost:3000/blog/post":
+        print(f"✅ Valid URL resolved correctly: {resolved}")
+    else:
+        print(f"❌ Valid URL failed to resolve correctly: {resolved}")
+        return False
+
+    # Case 2: URL not matching base
+    invalid_base_url = "https://evil.com/blog/post"
+    resolved = generate_embeddings.resolve_fetch_url(invalid_base_url, replacement_base, target_base)
+    if resolved is None:
+        print(f"✅ Invalid base URL correctly rejected")
```

Ruff (0.15.0)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)